### ISSUE TITLE
Invalid operation. Please check your configuration. OAuth2 authorization servers does notsupport /userinfo endpoint.

### ISSUE THREAD

**Issue by kuldip9060** [okta/samples-js-react-native#34]
Getting Error: "**Invalid operation. Please check your configuration. OAuth2 authorization servers does notsupport /userinfo endpoint**" When try to login with android emulator/device only this same thing is working fine with ios simulator/device and getting this error while try to access loggedin userinfo "**await getUser()**". 
here my configuration for reference.
```
oidc: {
    clientId: {My_Client_ID}',
    redirectUri: 'com.sampleapplication:/',
    endSessionRedirectUri: 'com.sampleapplication:/',
    discoveryUri: 'https://dev-493092.oktapreview.com/oauth2/default',
    scopes: ['openid', 'profile', 'offline_access'],
    requireHardwareBackedKeyStore: false,
  },
```

In Android/build.gradle file -> 
```
allprojects {
    repositories {
...
maven {
            url  "https://dl.bintray.com/okta/com.okta.android"
        }
...
}
}
```

In android/app/build.gradle file -> 

```
android {
defaultConfig {
...
manifestPlaceholders = [
                appAuthRedirectScheme: 'com.sampleapplication'
        ]
...
}
}
```

Please let me know what is happening wrong here. Thank You... K00L ;)

**Comment by swiftone** [comment]
Internal ref: OKTA-283789

**Comment by jdnichollsc** [comment]
Our current workaround is this:
```
const initializeOkta = () => {
  const oidc = {
    clientId: config.OKTA_CLIENT_ID,
    redirectUri: getDeepLink('callback'),
    endSessionRedirectUri: getDeepLink('logout'),
    discoveryUri: config.OKTA_ISSUER_URL + Platform.select({
      android: '/.well-known/openid-configuration',
      ios: ''
    }),
    scopes: [
      'openid',
      'profile',
      'offline_access',
      'user'
    ],
    requireHardwareBackedKeyStore: false
  }
  return createConfig({
    clientId: oidc.clientId,
    redirectUri: oidc.redirectUri,
    endSessionRedirectUri: oidc.endSessionRedirectUri,
    discoveryUri: oidc.discoveryUri,
    scopes: oidc.scopes,
    requireHardwareBackedKeyStore:
      oidc.requireHardwareBackedKeyStore
  })
}
```

**Comment by shuowu** [comment]
Another workaround is you can try to set the `discoveryUri` without `/oauth2/default`, then the SDK should be able to configure it automatically.

**Comment by jdnichollsc** [comment]
That workaround don't work for us because we have a custom issuerUrl, but thanks!

**Comment by swiftone** [comment]
@kuldip9060 - Until we find a better solution, have you been able to use one of the workarounds shown by @shuowu or @jdnichollsc?

**Comment by shihabpadath** [comment]
I also tried many ways, finally it worked suggested by @shuowu

**Comment by JayNewstrom** [comment]
This was an issue with the way the [Android SDK](https://github.com/okta/okta-oidc-android) handled discovery URIs. I confirmed the fix by adding `implementation 'com.okta.android:oidc-androidx:1.0.16'` to the sample apps `build.gradle`.

When we release the next version of the [React Native SDK](https://github.com/okta/okta-oidc-js/tree/master/packages/okta-react-native) it'll point to the fix in the Android SDK, and I'll close this issue then.

**Comment by JayNewstrom** [comment]
This is fixed in `1.4.4`.
